File: java/BingSpellCheck/src/main/java/BingSpellCheckQuickstart.java
import com.microsoft.azure.cognitiveservices.language.spellcheck.BingSpellCheckAPI;
import com.microsoft.azure.cognitiveservices.language.spellcheck.BingSpellCheckManager;
import com.microsoft.azure.cognitiveservices.language.spellcheck.models.SpellCheck;
import com.microsoft.azure.cognitiveservices.language.spellcheck.models.SpellCheckerOptionalParameter;
import com.microsoft.azure.cognitiveservices.language.spellcheck.models.SpellingFlaggedToken;
import com.microsoft.azure.cognitiveservices.language.spellcheck.models.SpellingTokenSuggestion;

import java.util.List;

/**
 * Sample code for spell checking using Bing Spell Check, an Azure Cognitive Service.
 *  - Spell check "Bill Gatas" with market and mode settings and print out the flagged tokens and suggestions.
 */
public class BingSpellCheckQuickstart {

    public static void main(String[] args) {
        try {
            // Authenticate - create a client
            final String subscriptionKey = System.getenv("BING_SPELL_CHECK_SUBSCRIPTION_KEY");
            BingSpellCheckAPI client = BingSpellCheckManager.authenticate(subscriptionKey);

            spellCheck(client);

        } catch (Exception e) {
            System.out.println(e.getMessage());
            e.printStackTrace();
        }
    }

    public static void spellCheck(BingSpellCheckAPI client) {
        System.out.println();
        String query = "bill Gatas was ehre today";

        SpellCheck result = client.bingSpellCheckOperations().spellChecker()
            .withText(query)
            .withMode("proof")
            .withMarket("en-us")
            .execute();

        System.out.println("Original query: \n" + query);
        System.out.println();

        // SpellCheck Results
        if (result.flaggedTokens().size() > 0) {

            // Get all misspelled words (tokens)
            System.out.println("Misspelled words: ");
            for (SpellingFlaggedToken token : result.flaggedTokens()) {
                System.out.println(token.token());
            }
            System.out.println();

            // Get suggestions for correct spellings
            System.out.println("Suggested corrections: ");
            for (SpellingFlaggedToken token : result.flaggedTokens()) {
                for (int i = 0; i < token.suggestions().size(); i++ ) {
                    System.out.printf(token.suggestions().get(i).suggestion() 
                        + " with confidence score " + "%.2f%n", token.suggestions().get(i).score());
                }
            }
        }
        System.out.println();
    }
}

File: dotnet/BingSpellCheck/BingSpellCheckQuickstart.cs
using System;
using System.Threading.Tasks;
using Microsoft.Azure.CognitiveServices.Language.SpellCheck;
using Microsoft.Azure.CognitiveServices.Language.SpellCheck.Models;

/*
 * This Bing Spell Check quickstart takes in some misspelled words and suggests corrections.
 * 
 * Prerequisites:
 * - Copy/paste this code into your Program.cs of a new Console App project in Visual Studio.
 * - Add your Bing Spell Check subscription key to your environment variables, using BING_SPELL_CHECK_SUBSCRIPTION_KEY as variable names.
 * - Install the NuGet package: Microsoft.Azure.CognitiveServices.Language.SpellCheck
 *   
 * References:
 * - Bing Spell Check documentation: https://docs.microsoft.com/en-us/azure/cognitive-services/bing-spell-check/index
 * - Dotnet SDK: https://docs.microsoft.com/en-us/dotnet/api/overview/azure/cognitiveservices/client/bingspellcheck?view=azure-dotnet
 * - API: https://docs.microsoft.com/en-us/rest/api/cognitiveservices-bingsearch/bing-spell-check-api-v7-reference
 */

namespace BingSpellCheckQuickstart
{
	class Program
	{
		static void Main(string[] args)
		{
			string query = "bill Gatas was ehre"; // Bill Gates was here

			// Authenticate
			string key = "PASTE_YOUR_SPELL_CHECK_SUBSCRIPTION_KEY_HERE";

			var client = new SpellCheckClient(
				new ApiKeyServiceClientCredentials(key));

			// Call API to check spelling of query
			checkSpelling(client, query).Wait();

		}

		public static async Task checkSpelling(SpellCheckClient client, string query)
		{
			var result = await client.SpellCheckerAsync(text: query, mode: "proof");

			Console.WriteLine("Original query: \n" + query);
			Console.WriteLine();
			Console.WriteLine("Misspelled words:");
			foreach (SpellingFlaggedToken token in result.FlaggedTokens)
			{
				Console.WriteLine(token.Token);
			}

			Console.WriteLine();
			Console.WriteLine("Suggested corrections:");
			foreach (SpellingFlaggedToken token in result.FlaggedTokens)
			{
				foreach(SpellingTokenSuggestion suggestion in token.Suggestions)
				{
					// Confidence values range from 0 (none) to 1.0 (full confidence)
					Console.WriteLine(suggestion.Suggestion + " with confidence " + Math.Round((decimal)suggestion.Score, 2));
				}
			}
		}
	}
}
